test/unit-tests/function/numeric/solveODE.test.js

```diff
@@ -0,0 +1,189 @@
+import assert from 'assert'
+import math from '../../../../src/defaultInstance.js'
+
+const solveODE = math.solveODE
+const matrix = math.matrix
+const subtract = math.subtract
+const exp = math.exp
+const min = math.min
+const max = math.max
+const abs = math.abs
+const tol = 1e-3
+const smaller = math.smaller
+const smallerEq = math.smallerEq
+const largerEq = math.largerEq
+const multiply = math.multiply
+const dotMultiply = math.dotMultiply
+const divide = math.divide
+const diff = math.diff
+const unit = math.unit
+const map = math.map
+const transpose = math.transpose
+
+function withinTolerance (A, B, tol) {
+  const maxAbsError = max(abs(subtract(A, B)))
+  return smaller(maxAbsError, tol)
+}
+
+describe('solveODE', function () {
+  function f (t, y) { return y }
+  function exactSol (T, y0) { return dotMultiply(y0, map(transpose([T]), exp)) } // this is only valid for y' = y
+  const tspan = [0, 4]
+  const y0 = [1, 2]
+
+  it('should throw an error if the first argument is not a function', function () {
+    assert.throws(function () {
+      const sol = solveODE('notAFunction', tspan, y0)
+      assert.deepStrictEqual(
+        withinTolerance(sol.y, exactSol(sol.t, y0), tol),
+        true
+      )
+    }, /TypeError: Unexpected type of argument in function solveODE \(expected: function.*/)
+  })
+
+  it('should throw an error if the second argument is not an array like', function () {
+    assert.throws(function () {
+      const sol = solveODE(f, 'NotAnArrayLike', y0)
+      assert.deepStrictEqual(
+        withinTolerance(sol.y, exactSol(sol.t, y0), tol),
+        true
+      )
+    }, /TypeError: Unexpected type of argument in function solveODE \(expected: Array or Matrix,.*/)
+  })
+
+  it('should throw an error if the second argument does not have a second member', function () {
+    assert.throws(function () {
+      const wrongTSpan = [tspan[0]]
+      const sol = solveODE(f, wrongTSpan, y0)
+      assert.deepStrictEqual(
+        withinTolerance(sol.y, exactSol(sol.t, y0), tol),
+        true
+      )
+    }, /TypeError: Unexpected type of argument in function.*/)
```

Comment:
Since `typed-function` is limited and you cannot define for example `tspan` as an Array with 2 numeric values, the error messages can be quite vague. Can you implement some manual validation on the inputs, so the function gives meaningful errors like "tspan" must be an Array containing two numbers [tStart, tEnd]?

Reply:
It makes sense, I will work on it,

I'm also concerned as tspan should be an array with either two numeric values or two units.

And they also need to match with `options.firstStep`, `options.maxStep` and `options.minStep` if provided.

Reply:
Now it throws an error regarding span